### ISSUE TITLE
Feature request: Add support for LibreFM / generic audio scrobbling APIs in settings

### ISSUE THREAD

**Issue by AntsyBoi** [Mastermindzh/tidal-hifi#513]
I use Libre.FM to track my listening habits, and there is currently no great way to track my tidal plays with LibreFM unless I do it in a browser with an extra extension installed. LibreFM uses the standard Audioscrobbler API and requires just a username and password to sign in. Alternatively, just making a generic scrobbler option in settings that for scrobblers compatible with that API would be a nice addition, as this would allow for using LibreFM or any other audioscrobbler without having to manually add support for each one.

**Comment by Mastermindzh** [comment]
I'm intrigued about this "Audioscrobbler API", you talk about it as if it's something shared between various solutions and not just a Libre.fm thing. Could you point me to some docs on that?

Either way I'll be happy to look into this at some time in the (hopefully near) future.

**Comment by AntsyBoi** [comment]
Libre.FM (from what I could understand from the weirdly limited documentation you can find with a quick search) uses the same protocol for scrobbling as LastFM and so most LastFM documentation on it should also apply. [https://www.last.fm/api/scrobbling](https://www.last.fm/api/scrobbling)

I don't know if this would help but an implementation of a generic audio scrobbler exists within Deadbeef's lastfm plugin, which also has support for LibreFM and might be a good reference [https://github.com/DeaDBeeF-Player/deadbeef/blob/master/plugins%2Flastfm%2Flastfm.c](https://github.com/DeaDBeeF-Player/deadbeef/blob/master/plugins%2Flastfm%2Flastfm.c) (I don't know how much that could help with developing it, but it is just a previous implementation of basically exactly what I'm describing)

**Comment by Mar0xy** [comment]
"You talk about it as if it's something shared between various solutions and not just a Libre.fm thing"

Actually it is basically just Last.FM's API which Libre.fm (back then gnu.fm) took and modified and then named generic audio scrobbler while this API works for basically all clients that I know of which are Maloja, Libre.FM and Last.FM it won't work for listenbrainz in turn maloja also supports listenbrainz api but libre.fm doesn't.

TL:DR Audio Scrobbling API's are a mess and I think there is no actual "generic" API

**Comment by espensa** [comment]
Tbf the generic API was the old Audioscrobbler site that was Richard Jones's university project in 2002 before it merged with Last.fm, apart from probably some fixes I don't think the API has really changed since then, and basically the same across other scrobblers. I can't believe my last.fm history goes back 23 years lol. It would be nice if you would add this feature, as the Tidal integration, and the Neptune plugins all seem to have issues

**Comment by espensa** [comment]
Your ScrobbleDelay is a bit short at least according to last.fm's definition: "A track should only be scrobbled when the following conditions have been met:

    The track must be longer than 30 seconds.
    And the track has been played for at least half its duration, or for 4 minutes (whichever occurs earlier.)

As soon as these conditions have been met, the scrobble request may be sent at any time. It is often most convenient to send a scrobble request when a track has finished playing."

**Comment by stafio** [comment]
> Bump
> 
> The tidal app on windows has last.fm scrobbling natively, so I was hoping that this would too

The lastfm integration does show on https://listen.tidal.com/settings now and this player is "The web version of listen.tidal.com running in electron...", so hopefully it comes with the next update
